File: algorithms/122.Best_Time_to_Buy_and_Sell_Stock_II/Best_Time_to_Buy_and_Sell_Stock_II.py
class Solution:
    def maxProfit(self, prices):
        profit = 0
        for i in range(len(prices) - 1):
            profit += max(0, prices[i + 1] - prices[i])     
        return profit

if __name__ == "__main__":
    result = Solution().maxProfit([3, 2, 1, 4, 2, 5, 6])
    print(result)

File: algorithms/122.Best_Time_to_Buy_and_Sell_Stock_II/best-time-to-buy-and-sell-stock-ii.go
package problem0122

func maxProfit(prices []int) int {
	var max int = 0
	for i := 1; i < len(prices); i++ {
		if prices[i]-prices[i-1] > 0 {
			max += prices[i] - prices[i-1]
		}
	}
	return max
}
